File: Node/TypeORM_ORM/src/bll/services/territoryService.ts
import  { TerritoryRepository } from '../../dal/repositories/territoryRepository'
import { Territories } from '../../dal/entities/Territories.js';

export class TerritoryService {
  private territoryRepository = new TerritoryRepository();
  async getAll() {
    return await this.territoryRepository.getAll();
  }

  async getById(id: string) {
    return await this.territoryRepository.getById(id);
  }

  async create(data: Partial<Territories>) {
    return await this.territoryRepository.create(data);
  }

  async update(id: string, data: Partial<Territories>) {
    return await this.territoryRepository.update(id, data);
  }

  async delete(id: string) {
    return await this.territoryRepository.delete(id);
  }
}

File: Go/GORM_ORM/bll/services/territoryService.go
package services

import (
	"GORM_ORM/dal/entities"
	"GORM_ORM/dal/repositories"
)

type TerritoryService interface {
	GetAll() ([]entities.Territory, error)
	GetByID(id string) (*entities.Territory, error)
	Create(territory *entities.Territory) error
	Update(territory *entities.Territory) error
	Delete(id string) error
}

type territoryService struct {
	repo repositories.TerritoryRepository
}

func NewTerritoryService(repo repositories.TerritoryRepository) TerritoryService {
	return &territoryService{repo: repo}
}

func (s *territoryService) GetAll() ([]entities.Territory, error) {
	return s.repo.GetAll()
}

func (s *territoryService) GetByID(id string) (*entities.Territory, error) {
	return s.repo.GetByID(id)
}

func (s *territoryService) Create(territory *entities.Territory) error {
	return s.repo.Create(territory)
}

func (s *territoryService) Update(territory *entities.Territory) error {
	return s.repo.Update(territory)
}

func (s *territoryService) Delete(id string) error {
	return s.repo.Delete(id)
}
